packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -194,19 +194,19 @@ export class InternalModelManager {
     this.capitalizedApiIdentifier = camelize(apiIdentifier);
   }
 
-  async findOne(id: string, throwOnEmptyData = true): Promise<GadgetRecord<any>> {
+  async findOne(id: string, throwOnEmptyData = true): Promise<GadgetRecord<unknown>> {
     const response = await this.connection.currentClient.query(internalFindOneQuery(this.apiIdentifier), { id }).toPromise();
     const assertSuccess = throwOnEmptyData ? assertOperationSuccess : assertNullableOperationSuccess;
     const result = assertSuccess(response, ["internal", this.apiIdentifier]);
-    return hydrateRecord(response, result);
+    return await hydrateRecord(response, result);
```

Comment:
Do we need `await` here? It doesn't look like `hydrateRecord` is an async function. If it's because you know (or suspect) that `hydrator` might be async, then should we also change `hydrateRecord` to be async + await the hydrator call?

Reply:
Interesting... I added the `await` because ESLint told me to

```
/Users/scott/Code/gadget/js-clients/packages/api-client-core/src/InternalModelManager.ts
  201:12  error  Returning an awaited promise is required in this context  @typescript-eslint/return-await
```

but you're right, it's not an async function so I'm not exactly sure why ESLint is complaining... I'm gonna look into this more.

Reply:
Waaaa??? That's so weird! It looks like it's because of the change from `any` to `unknown`. I'm fine with leaving it in, just a little confused why eslint thinks it's needed. 🤷🏻‍♂️